### PR TITLE
Enhance twitter demoapp

### PR DESCRIPTION
**Pull request by ading1977**
This PR makes the following enhancements to the twitter app to better support horizontal scale:

* Allow periodic GRPC channel reconnect to maintain load balancing across multiple service replicas. This is achieved by setting `grpc.max_connection_age_ms` on the GRPC channel on the server side (i.e., `tweet`, `user` and `friend` services)
* Implement client side retries from `http` service to `tweet`/`user`/`friend` services, this is needed because the server can disconnect GRPC channel periodically to achieve load balancing. We use the `retrying` package to retry up to 5 times with 100ms interval
* Allow customizing the following through command line options:
  * `grpc.max_connection_age_ms` (for channel reconnect, default to `None` meaning no reconnect), for example:
   ```
  containers:
  - args:
    - --grpc-max-connection-age-ms 5000
    env:
    - name: CASSANDRA_HOST_ADDRESS
      value: cassandra
    image: ading1977/twitter-cass-tweet:latest
    imagePullPolicy: Always
    name: twitter-cass-tweet
   ```
  * logging level
  * database query timeout in seconds (default to 5 seconds, instead of the 0.5 seconds hard coded now, to tolerate longer queries)
  * database query retry count (default to 5)
* Refactor code to share common logic (implemented in `common_service`) including logging configuration and GRPC server creation
* Add `Dockerfile` to build base image, and install `instana` packages into base image
* Activate `instana` package

Also, I am setting the repo to `turbonomic` instead of `vmturbo`, and will push image to `turbonomic` after the review is approved.

Will submit a separate PR to modify the deployment of demoapp when images are pushed

### CODE REVIEW

**Review by chlam4 — APPROVED** [review]
/lgtm, thanks @ading1977!!

File: friend_service/friend_service_grpc.py [hunk 1]
```diff
@@ -1,13 +1,18 @@
-from concurrent import futures
+import instana
```
Comment (esara) [inline friend_service/friend_service_grpc.py:1]: all looks good, why do we need to import instana?

File: http_server/http_server.py [hunk 2]
```diff
@@ -1,19 +1,62 @@
+import instana
```
Comment (esara) [inline http_server/http_server.py:1]: same question as above

**Review by enlinxu — APPROVED** [review]
/lgtm

File: friend_service/friend_service_grpc.py (hunk 1 above)
Comment (ading1977) [reply friend_service/friend_service_grpc.py:1]: This is to manually activate `instana` instrumentation. It won't take effect if instana agent is not installed. The Instana auto-injection doesn't seem to work for me, so I follow their instruction to enable manual injection.

File: http_server/http_server.py (hunk 2 above)
Comment (ading1977) [reply http_server/http_server.py:1]: Same as above.

**Pull request merged**